src/org/pentaho/metastore/stores/xml/XmlMetaStoreElementType.java

```diff
@@ -180,4 +180,10 @@ public void setFilename( String filename ) {
     this.filename = filename;
   }
 
+  private DocumentBuilderFactory createSafeDocumentBuilderFactory() throws ParserConfigurationException {
```

Comment:
This is needless code duplication. XmlMetaStoreElementType and XmlMetaStoreElement should use a common createSafeDocumentBuilderFactory method, either in package, or out. But we certainly shouldn't have the same exact code in two different classes in the same exact package.